reporters/GlobalSummaryReporter.js

```diff
@@ -0,0 +1,476 @@
+    await this._generateLiquidationStats();
+    console.groupEnd();
+
+    // 4. Dispute stats table
+    console.group();
+    console.log(bold("Dispute summary stats"));
+    await this._generateDisputeStats();
+    console.groupEnd();
+
+    // 5. DVM stats table
+    console.group();
+    console.log(bold("DVM summary stats"));
+    await this._generateDvmStats();
+    console.groupEnd();
+  };
+
+  _generateSponsorStats = async () => {
+    const { toBN, toWei } = this.web3.utils;
+
+    let allSponsorStatsTable = {};
+
+    if (this.newSponsorEvents.length === 0) {
+      console.log(dim("\tNo positions have been created for this EMP."));
+      return;
+    }
+
+    // - Lifetime # of unique sponsors.
+    const uniqueSponsors = {};
+    const currentUniqueSponsors = this.empClient.getAllPositions();
+    for (let event of this.newSponsorEvents) {
+      uniqueSponsors[event.sponsor] = true;
+    }
+    allSponsorStatsTable["# of unique sponsors"] = {
+      cumulative: Object.keys(uniqueSponsors).length,
+      current: Object.keys(currentUniqueSponsors).length
+    };
+
+    // - Cumulative collateral deposited into contract
+    let collateralDeposited = toBN("0");
+    let collateralDepositedPeriod = toBN("0");
+    for (let event of this.collateralDepositEvents) {
+      collateralDeposited = collateralDeposited.add(toBN(event.returnValues.value));
+      if (event.blockNumber >= this.startBlockNumberForPeriod) {
+        collateralDepositedPeriod = collateralDepositedPeriod.add(toBN(event.returnValues.value));
+      }
+    }
+    allSponsorStatsTable["collateral deposited"] = {
+      cumulative: this.formatDecimalString(collateralDeposited),
+      [this.periodLabelInHours]: this.formatDecimalString(collateralDepositedPeriod),
+      current: this.formatDecimalString(this.totalPositionCollateral)
+    };
+
+    // - Cumulative collateral withdrawn from contract
+    let collateralWithdrawn = toBN("0");
+    let collateralWithdrawnPeriod = toBN("0");
+    for (let event of this.collateralWithdrawEvents) {
+      collateralWithdrawn = collateralWithdrawn.add(toBN(event.returnValues.value));
+      if (event.blockNumber >= this.startBlockNumberForPeriod) {
+        collateralWithdrawnPeriod = collateralWithdrawnPeriod.add(toBN(event.returnValues.value));
+      }
+    }
+    allSponsorStatsTable["collateral withdrawn"] = {
+      cumulative: this.formatDecimalString(collateralWithdrawn),
+      [this.periodLabelInHours]: this.formatDecimalString(collateralWithdrawnPeriod)
+    };
+
+    // - Net collateral deposited into contract:
+    let netCollateralWithdrawn = collateralDeposited.sub(collateralWithdrawn);
+    let netCollateralWithdrawnPeriod = collateralDepositedPeriod.sub(collateralWithdrawnPeriod);
+    allSponsorStatsTable["net collateral deposited"] = {
+      cumulative: this.formatDecimalString(netCollateralWithdrawn),
```

Comment:
Intuitively, this should be the same as the result of `await this.empContract.methods.pfc()`, right? I'm not suggesting you compute it that way (the fewer contract calls, the better, IMO) -- just want to make sure I'm understanding what this value is representing.

Reply:
The `netCollateralWithdrawn` is simply equal to the difference between `Transfers` in and out.

The `pfc()` can be less than the `netCollateralWithdrawn` if there is a pending liquidation. This is why I display the `current collateral liquidated` value in the Liquidation stats table.

WDYT about a better way to display this?

Reply:
`pfc()` includes pending liquidations, though, right?

Reply:
>pfc() includes pending liquidations, though, right?

correct.

I misunderstood and updated my prior comment

Reply:
This all looks good to me -- no suggestions for improvement.